### PR TITLE
Use api constraint class to handle API versionong through headers

### PR DESCRIPTION
**Pull request by dmitrytrager**
This PR introduces `ApiConstraint` class that will allow switching between api versions using http headers.

Also, couple of bin script were added. `bin/ci` can be used to trigger checks on TravisCI and it also calls `bin/quality` which, in it's turn, introduces code quality checks.

Rails version and sources for recently introduced gems were updated to prevent some vulnerability issues.

### CODE REVIEW

File: app/controllers/api/api_controller.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class Api::ApiController < ActionController::Base
+class Api::ApiController < ActionController::API
```
Comment (dmitrytrager) [inline app/controllers/api/api_controller.rb]: @t-harps @GarySiu are there specific reasons to inherit this and below one from Base controller and not from API ?

Comment (t-harps) [inline app/controllers/api/api_controller.rb]: I don't think so, @GarySiu ?

Comment (GarySiu) [inline app/controllers/api/api_controller.rb]: No it was the default from when it was generated and never touched since

**Comment by t-harps** [comment]
How does the versioning actually work? trying to figure out how to make an API request on this branch and failing, do I need a new header?

**Comment by t-harps** [comment]
Never mind figured it out!

**Comment by dmitrytrager** [comment]
@t-harps It defaults to v1. You need headers only if you want another version.

**Comment by dmitrytrager** [comment]
please, review

**Pull request merged**